### PR TITLE
Add filter from in test overview

### PR DESCRIPTION
**Pull request by Martchus**

### CODE REVIEW

**Comment by okurz** [comment]
see screenshot:
openqa_tests_overview_filter

seems to work as expected, IMHO only needs design improvements, e.g. the obvious overlap as visible in screenshot.

**Comment by Martchus** [comment]
I also noticed that the design is messed in Firefox. (It looks good in Chromium.)

**Comment by Martchus** [comment]
I fixed the design issues in Firefox and it should work.

Should I outsource the JavaScript and CSS code to extra files?

Maybe someone has a more elegant solution for adding hidden input elements and setting initial form status according current query parameter?

@coolo I assume you prefer
```
$('#filter-form').append($('<input value="' + val + '" name="' + key + '" type="hidden">'));
```
over
```
hiddenInput = document.createElement('input');
hiddenInput.type = 'hidden';
hiddenInput.name = key;
hiddenInput.value = val;
document.getElementById('filter-form').appendChild(hiddenInput);
```
? Just because here I find the code using plain DOM easier to read.

Should I write tests for it? (It is just a form and a few lines JavaScript.)

**Comment by okurz** [comment]
On Wednesday 07 September 2016 01:12:36 Martchus wrote:
> Should I write tests for it? (It is just a form and a few lines JavaScript.)

yes, of course :-)

**Comment by coolo** [comment]
```
var input = $('<input/>');
input.attr('value', val);
input.attr('name', key);
input.attr('hidden', true);
$('#filter-form').append(input);

**Comment by Martchus** [comment]
I took the jQuery code for consistency and added tests.

File: templates/test/overview.html.ep [hunk 1]
```diff
@@ -83,4 +83,80 @@
...
+    margin-right: 20px;
+    font-weight: normal;
+}
+#filter-form .form-group input {
+    margin-right: 2px;
+    padding-right: 0px;
+}
+</style>
+<script type="text/javascript">
+// ensure TODO is false by default because the Browser auto remembers previous state
```
Comment (coolo) [inline templates/test/overview.html.ep]: no script and no style in templates please. They are not minified and not cached this way

Comment (Martchus) [inline templates/test/overview.html.ep]: Ok, that was one of my questions.

File: assets/javascripts/overview.js [hunk 2]
```diff
@@ -35,5 +35,41 @@ function setupOverview() {
...
+                input.attr('name', key);
+                input.attr('hidden', true);
+                $('#filter-form').append(input);
+            }
+        }
+    }
+    
+    // set enabled/disabled state of checkboxes (according to current filter)
+    var badges = ['passed', 'incomplete', 'softfailed', 'failed', 'scheduled', 'running', 'none', 'unknown'];
+    for(var i = 0; i < badges.length; ++i) {
```
Comment (coolo) [inline assets/javascripts/overview.js]: http://api.jquery.com/jquery.each/ can be quite handy

File: assets/javascripts/overview.js [hunk 3]
```diff
@@ -35,5 +35,41 @@ function setupOverview() {
 	    icon.fadeTo('slow', 0.5).fadeTo('slow', 1.0);
 
 	});
+
+    // ensure TODO is false by default because the Browser auto remembers previous state
+    $('#filter-todo').prop('checked', false);
+    
+    // find specified results
+    var varPairs = window.location.search.substring(1).split('&');
```
Comment (coolo) [inline assets/javascripts/overview.js]: stackoverflow? :)

Comment (Martchus) [inline assets/javascripts/overview.js]: Used the same approach already in other projects.

I searched before whether jQuery provides a convenient way, but only via a plug-in and I decided it is not worth using additional packages for this simple task.

File: assets/javascripts/overview.js (hunk 2 above)
Comment (Martchus) [inline assets/javascripts/overview.js]: It would save the `var badge = badges[i];`. I'll maybe use it next time.

**Pull request merged**